pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/BaseClusterIntegrationTest.java

```diff
@@ -941,58 +1113,106 @@ public File getSchemaFile() {
         .getResource("On_Time_On_Time_Performance_2014_100k_subset_nonulls.schema").getFile());
   }
 
+  /**
+   * NOTE:
+   * <p>
+   * For queries with <code>LIMIT</code> or <code>TOP</code>, need to remove limit or add <code>LIMIT 10000</code> to
+   * the H2 SQL query because the comparison only works on exhausted result with at most 10000 rows.
+   * <ul>
+   *   <li>Eg. <code>SELECT a FROM table LIMIT 15 -> [SELECT a FROM table LIMIT 10000]</code></li>
+   * </ul>
+   * <p>
+   * For queries with multiple aggregation functions, need to split each of them into a separate H2 SQL query.
+   * <ul>
+   *   <li>Eg. <code>SELECT SUM(a), MAX(b) FROM table -> [SELECT SUM(a) FROM table, SELECT MAX(b) FROM table]</code></li>
+   * </ul>
+   * <p>
+   * For group-by queries, need to add group-by columns to the select clause for H2 SQL query.
+   * <ul>
+   *   <li>Eg. <code>SELECT SUM(a) FROM table GROUP BY b -> [SELECT b, SUM(a) FROM table GROUP BY b]</code></li>
+   * </ul>
+   *
+   * @throws Exception
+   */
   @Test
-  public void testHardcodedQuerySet() throws Exception {
-    for (String query : getHardCodedQuerySet()) {
-      try {
-        LOGGER.debug("Trying to send query : {}", query);
-        runQuery(query, Collections.singletonList(query.replace("'mytable'", "mytable")));
-      } catch (Exception e) {
-        LOGGER.error("Getting error for query : {}", query, e);
-      }
-
-    }
+  public void testHardcodedQueries()
```

Comment:
You are right, we should disable this one by default, this one is used to manually test some queries.